ocaml/libs/open-uri/open_uri.ml

```diff
@@ -53,6 +53,20 @@ let with_open_uri uri f =
              (Uri.to_string uri)
           )
   )
+  | Some "https" -> (
+    match (Uri.host uri, Uri.port uri) with
+    | Some host, Some port ->
+        Stunnel.with_connect ~verify_cert:(Stunnel_client.pool ()) host port (fun s ->
+          f Safe_resources.Unixfd.(!(s.Stunnel.fd)))
```

Comment:
that is not really a safe way to pass that fd, the wrapper might decide to close it at some point, can we instead pass it the `s.Stunnel.fd` directly and let it do the unwrapping internally?

Reply:
Do you mean that `Safe_resources.Unixfd.(!)` will close it?

I'll take a look, but it comes down to the semantics of the existing function, and the way `f` is used by callers now. If we'd push down the Safe_resources thing, then we'll also need to do that for the other cases.

Reply:
The best way forward seems to be to adopt Safe_resources for all fds in this function, but that would be a much bigger change. At least `Stunnel.with_connect` does not close the fd as far as I can see.